### ISSUE TITLE
Fix Trezor connectivity upon firmware upgrade

### ISSUE THREAD

**Issue by markmhx** [blockstack/stacks-wallet#254]
The ability to authenticate the current Stacks Wallet with a Trezor device current wallet breaks regardless of model whenever a user upgrades their Trezor firmware to any version above ~1.6~ 1.8.3.

@ryanarndtcm has been encouraging reporting users to downgrade their firmware to that version, but this guidance doesn't appear to resolve the problem for most given the difficulty of downgrading.

The most decisive fix would entail rewriting our Trezor support holistically to support newer firmware versions. However, that would pose a large engineering effort ahead of Stacks 2.0 that would get rendered null by the new wallet, which will likely not support Trezor at all and require all Trezor users to migrate to Ledger or software mode anyway.

Given this context, it appears that the most effective short-term solution is an upgrade to the current wallet that permits all users to enter 12-word seeds in addition to 24-word ones. That would permit Trezor users to export their 12-word seeds and use them in software mode immediately. It'd have the added benefit of preparing them for transition to the new wallet.

The new wallet will also support 12-word seed entry as per https://github.com/blockstack/stacks-wallet/issues/253. However, given the large volume of unresolved support requests we currently receive concerning this firmware bug, it seems prudent to resolve it in the current version as well.

This bug has also appeared in related to reports of [missing Trezor Model T support](https://github.com/blockstack/stacks-wallet/issues/97). However, this bug appears to be separate since that other bug involves incompatible versions of Trezor Connect.

**Comment by markmhx** [comment]
@ryanarndtcm can you confirm here how often you receive reports about this specific bug (and not the linked Model T one), as well as how often you're able to help users resolve it with your firmware downgrade guidance?

@yknl @agraebe I'm assigning this to you for assessing any tradeoffs you face with tackling this fix for the current wallet given DevX priorities in general and https://github.com/blockstack/stacks-wallet/issues/131 in particular.

The UserX team will assume responsibility for all Trezor fixes and migration needs for the _new_ wallet (e.g. as per https://github.com/blockstack/stacks-wallet/issues/253 and https://github.com/blockstack/stacks-wallet/issues/248).

**Comment by yknl** [comment]
Correction: Trezor firmware v1.8.3 and below are compatible.

**Comment by agraebe** [comment]
> However, given the large volume of unresolved support requests we currently receive concerning this firmware bug, it seems prudent to resolve it in the current version as well.

thanks @markmhx. Just to verify, the request is specifically to support a 12-word seed in the old wallet?

> However, that would pose a large engineering effort ahead of Stacks 2.0 that would get rendered null by the new wallet, which will likely not support Trezor at all and require all Trezor users to migrate to Ledger or software mode anyway.

@yknl and I both feel like the ROI might be too low given the arguments above. It would further delay the announced BTC withdrawal feature (#131).

**Comment by agraebe** [comment]
Following up to keep everyone in the loop. @markmhx @aulneau @yknl and @ryanarndtcm had ad hoc conversations today to get to the bottom of this. 

**TL;DR**: We will add instructions for Trezor users to the current wallet and observe issues coming in throughout the next sprint. With a better understanding of the reach and impact of this issue, we will make a decision during the next DevX sprint plan (~2 weeks from now).

----

Action items:
- [ ] @aulneau Add disclaimer in the current wallet to help Trezor users (#256) 
- [ ] @agraebe Create a page that explains the two options we will suggest (https://github.com/blockstack/docs/issues/782)
- [ ] @yknl to verify that thw two options work (esp the Trezor to Ledger switch)
- [ ] @ryanarndtcm Share all Trezor-related issues with in a group chat for more visibility

---

A few more notes from the conversations:
- **Effort**: If we were to implement the 12-word seed option, we'd need to update mostly UI elements (in at least two places). This change, together with the BTC withdrawal feature, would likely reason a security audit
- **Dependency**: We will decouple this issue and move ahead with a release of the current wallet with the BTC withdrawal feature (#131)
- **Reach**: Trezor-related issues started to pick up in mid-July, with ~40 tickets so far (each from a different individual). That's < 1 ticket/day, so lower that what we thought initially, but non-negligible. It is difficult to predict reach based on the recent trend change
- **Importance**: Relatively speaking, the Trezor issues are considered much less important compared to the BTC withdrawal feature
- **Usefulness**: There seems to be some hesitancy to enable the software wallet usage due to security implications
